chi/share.py

```diff
@@ -0,0 +1,138 @@
+from .clients import manila
+
+from manilaclient.exceptions import NotFound
+
+__all__ = [
+    'create_share',
+    'delete_share',
+    'extend_share',
+    'get_access_rules',
+    'get_share',
+    'get_share_id',
+    'list_shares',
+    'shrink_share',
+]
+
+
+def _get_default_share_type_id():
+    # we only support one share type - cephfsnfstype
+    share_types = manila().share_types.list()
+    if not share_types:
+        raise ValueError("No share types found")
+    elif len(share_types) > 1:
+        raise ValueError("Multiple share types found")
+    return share_types[0].id
+
+
+def create_share(size, name=None, description=None, metadata=None,
+                 is_public=False):
+    """Create a share.
+
+    Args:
+        size (int): size in GiB.
+        name (str): name of new share.
+        description (str): description of a share.
+        is_public (bool): whether to set share as public or not.
+
+    Returns:
+        The created share.
+    """
+    share = manila().shares.create(
+        share_proto="NFS",
+        size=size,
+        name=name,
+        description=description,
+        metadata=metadata,
+        share_type=_get_default_share_type_id(),
+        is_public=is_public
+    )
+    return share
+
+
+def delete_share(share):
+    """Delete a share.
+
+    Args:
+        share: either share object or text with its ID.
+    """
+    manila().shares.delete(share)
+
+
+def extend_share(share, new_size):
+    """Extend the size of the specific share.
+
+    Args:
+        share: either share object or text with its ID.
+        new_size: desired size to extend share to.
+    """
+    manila().shares.extend(share, new_size)
+
+
+def get_access_rules(share):
+    """Get access list to a share.
+
+    Args:
+        share: either share object or text with its ID.
+
+    Returns:
+        A list of access rules.
+    """
+    return manila().shares.access_list(share)
+
+
+def get_share(ref):
+    """Get a share by its ID or name.
+
+    Args:
+        ref (str): The ID or name of the share.
+
+    Returns:
+        The share matching the ID or name.
+
+    Raises:
+        NotFound: If the share could not be found.
+    """
+    try:
+        return manila().shares.get(ref)
+    except NotFound:
+        return manila().shares.get(get_share_id(ref))
```

Comment:
In order to avoid the second API call here, we could edit `get_share_id` to just return the ID if it is a UUID. I have seen this pattern in other OpenStack code. That way, we could just do 

```python
return manila().shares.get(get_share_id(ref))
```

Reply:
> In order to avoid the second API call here, we could edit `get_share_id` to just return the ID if it is a UUID. I have seen this pattern in other OpenStack code. That way, we could just do
> 
> ```python
> return manila().shares.get(get_share_id(ref))
> ```

What if the share has a UUID as its name?